lib/discover/index.js

```diff
@@ -0,0 +1,21 @@
+var url = require('url')
+
+module.exports.protocols = {
+    "hyperbahn:": require('./hyperbahn').discover,
+    "file:": require('./file').discover,
+    "ip:": require('./ip').discover,
+    // by default fallback on ip discovery
+    default: require('./ip').discover
+}
+
+// find a list of initial nodes to connect to based on the discover string
+module.exports.discover = function (discover, callback) {
+    var urlObj = url.parse(discover)
+    var handler = module.exports.protocols[urlObj.protocol];
```

Comment:
Makes sense. My initial idea was to populate the hash dynamically based on the file names. But than the colon showed up in the protocol when parsed, I decided to add a fallback for `ip` when the protocol was not recognized and it is not really needed to auto-discover the discovery (that is meta).

I'm thinking about implementing it with a switch.